keps/sig-multicluster/1645-multi-cluster-services-api/specification.md

```diff
@@ -0,0 +1,263 @@
+  - [2.4 - Records for a Multicluster Headless Service](#24---records-for-a-multicluster-headless-service)
+    - [2.4.1 - <code>A</code>/<code>AAAA</code> Records](#241----records)
+    - [2.4.2 - <code>SRV</code> Records](#242----records)
+    - [2.4.3 - <code>PTR</code> Records](#243----records)
+      - [Limitations](#limitations-1)
+    - [2.4.4 - Records that should NOT exist for a Multicluster Headless Service](#244---records-that-should-not-exist-for-a-multicluster-headless-service)
+<!-- /toc -->
+
+## 0 - About This Document
+
+This document is a specification for DNS-based Kubernetes service discovery for clusters implementing the [Multicluster Service API](README.md).
+
+## 1 - Schema Version
+
+This document describes version 1.0.0 of the schema.
+
+## 2 - Resource Records
+
+Any DNS-based service discovery solution for Kubernetes clusters implementing the Multicluster Services API must provide the resource records (RR) described below to be considered compliant with this specification.
+
+### 2.1 - Definitions
+
+This proposal is intended as an extension of the [cluster-local Kubernetes DNS specification](https://github.com/kubernetes/dns/blob/master/docs/specification.md), and inherits its definitions from section 2.1  with the addition of the following:
+
+hostname = as already defined in the [cluster-local Kubernetes DNS specification](https://github.com/kubernetes/dns/blob/master/docs/specification.md), this refers (in brief): in order of precedence, to a) the value of the endpoint's `hostname` field, or b) a unique, system-assigned identifier for the endpoint. Of importance to highlight is that since the [default hostname of an endpoint is the Pod's `metadata.name` field](https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields), this will likely often be the podname, but not always, and implementations must prefer a directly specified `hostname` value.
+
+clusterset = as defined in [KEP-1645: Multi-Cluster Services API](README.md): “A placeholder name for a group of clusters with a high degree of mutual trust and shared ownership that share services amongst themselves. Membership in a clusterset is symmetric and transitive. The set of member clusters are mutually aware, and agree about their collective association. Within a clusterset, [namespace sameness](https://github.com/kubernetes/community/blob/master/sig-multicluster/namespace-sameness-position-statement.md) applies and all namespaces with a given name are considered to be the same namespace.”
+
+`<clustersetzone>` = domain for multi-cluster services in the clusterset, which must be `clusterset.local`; as this may become configurable in the future, this specification refers to it by the placeholder `<clustersetzone>`, but per the MCS API it currently must be defined to be `clusterset.local`. 
```

Comment:
super nit.  You define `<clustersetzone>` and `<clusterset-ip>`.  why not `clusterset-zone` or `clustersetip`

Reply:
Fixed in [d2e046d](https://github.com/kubernetes/enhancements/pull/2577/commits/d2e046d7c5a5569bc2c831aa2450dbb2e3eaa56e)